All is relative

jgosmann opened the pull request:
Eliminates all absolute sizing from components and uses relative scaling/padding/margin based on the font-size. Also, makes the text zoom with everything else.

While doing this, I refactored the components to have less code duplication.

This is based on some intermediate stage of `graphics_overhaul`.

jgosmann commented:
This is just a first pass on refactoring the components. There are probably still a number of improvements possible.

tcstewar commented on nengo_viz/static/slidercontrol.js:
```diff
@@ -0,0 +1,209 @@
+VIZ.SliderControl = function(min, max) {
```
Should this file be ```viz_slidercontrol.js``` for consistency?

jgosmann replied:
For now this would make sense. But in general I would favor dropping the `viz` prefix in filenames altogether. It is not really informative because everything is prefixed with `viz` and gives only clutter when listing the files. If at all needed I would prefer to use a directory instead of prefixing the files.

It might make sense to also provide some more structure with some directories given the growing number of files, for example move all components to their own subdirectory.

tcstewar replied:
I totally agree with dropping the ```viz_*``` prefix on the files.  We'll put it on the list of things to do once it's all merged... 

And, in fact, if you want to leave this one the way it is (as kind of a marker that we should fix this in the future), I'm good with that too.

tcstewar commented:
the file ```.DS_Store``` needs to get removed, and then there's the whole rebasing to do, but other than that I like this refactoring.

jgosmann commented:
The `.DS_Store` file was committed accidentally in the `graphics_overhaul` branch which is partially part of this branch. I would suggest to merge that branch first and then rebase this one. But I could also rebase this one first (I don't think, I depend on anything particular from that branch.)

tcstewar commented:
> I would suggest to merge that branch first and then rebase this one.

Is that branch ready for merging?  I don't see a PR for it.....

jgosmann commented:
just made one: #180

jgosmann commented:
Rebased to `master` after `graphics_overhaul` has been merged. I might have broken some slider stuff in resolving the merge conflicts. Will test that tomorrow.

jgosmann commented:
Fixed the sliders and cleaned up the history. (Commits might be shown in the wrong order on Github, because they are shown by time stamp and not by parent.)

Remaining problems:
* [ ] The size of the sliders is incorrectly initialized on load.
* [ ] The font scaling is incorrectly initialized on load.

jgosmann commented:
One more to-do: Move the CSS styling of the slider from the JS to a CSS file. I don't think this prevents a merge because it is only about code organization.

tcstewar commented:
> I don't think this prevents a merge because it is only about code organization.

Looking great!  Now I just have to decide on which order to merge these large things into master in....  That'll be my job for tomorrow after a good night's sleep...  :)
